Dockerfile

```diff
@@ -19,28 +24,21 @@ RUN mkdir /projects ${HOME} && \
       chmod -R g+rwX ${f}; \
     done
 
-# odo and oc versions have to match the ones defined in https://github.com/redhat-developer/vscode-openshift-tools/blob/master/src/tools.json
```

Comment:
Hello, is there a reason to move it to an upper level ? (it reduces caching layer)
and change ENV format ? (it's a lint error ? )

Reply:
I mean, if each committer is applying his preferences on ENV params and cli options each time there is a bump in versions it might become a lot of changes for nothing (I have no preferences at all, just wanting the smallest changes each time)

Reply:
I was trying to collapse steps. I thought you'd approve of having all the ENVs declared in a single place instead of in several commands. Why define HOME and then do some work, and then define more ENVs later? It's just... smelly code.

Reply:
BTW if you want smaller changes, we should be working in master branch and branching from that for each new release, rather than creating an empty branch and adding new files. 

Compare:
https://github.com/che-dockerfiles/che-sidecar-openshift-connector/commits/0.1.1/Dockerfile (one commit)
https://github.com/che-dockerfiles/che-sidecar-openshift-connector/commits/0.1.2/Dockerfile (one commit)
(no git blame to diff versions and see why lines have changed)

vs...

https://github.com/eclipse/che-machine-exec/commits/7.7.0/Dockerfile (a smaller series of commits)
https://github.com/eclipse/che-machine-exec/commits/7.8.0/Dockerfile (a few more commits, with the ability to do git blame)

This "all files are new" approach means it's super difficult to know what's changed between releases. Why are we doing it that way? For obfuscation?

Reply:
I was referencing ENV with quotes vs ENV with spaces

Reply:
You may also look at https://github.com/eclipse/che/issues/15949

Reply:
Quotes are optional, but recommended (shellcheck, probably). 

But if you use spaces you can only do one ENV declaration per line. Using equal signs allow you to chain them into a single layer.